packages/website/src/components/inputs/Dropdown.tsx

```diff
@@ -1,28 +1,52 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import styles from '../OptionsSelector.module.css';
 import clsx from 'clsx';
 
-export interface DropdownProps {
-  readonly onChange: (value: string) => void;
-  readonly options: string[];
-  readonly value: string | undefined;
+export interface DropdownOption<T> {
+  readonly value: T;
+  readonly label: string;
+}
+
+export interface DropdownProps<T> {
+  readonly onChange: (value: T) => void;
+  readonly options: readonly (DropdownOption<T> | T)[];
+  readonly value: T | undefined;
   readonly name: string;
   readonly className?: string;
 }
 
-function Dropdown(props: DropdownProps): JSX.Element {
+function Dropdown<T extends boolean | string | number>(
+  props: DropdownProps<T>,
+): JSX.Element {
+  const [options, setOptions] = useState<DropdownOption<T>[]>([]);
+
+  useEffect(() => {
+    setOptions(
+      props.options.map(option =>
+        typeof option !== 'object'
+          ? { label: String(option), value: option }
+          : option,
+      ),
+    );
+  }, [props.options]);
```

Comment:
FYI - when deriving state from props, the recommended pattern is to do the "setState" inline in the render.

The problem with using an effect to do this is that it causes any prop changes to be applied later than desired.

https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops

Reply:
as i'm rebuilding `props.options` i can't do simple comparison, `if (row !== prevRow) {`, thus i will have to do shallow equal check from your example, and at that point its actually better to not use check at all

than you are recommending me to do:
```ts
const options = props.options.map(option =>
        typeof option !== 'object'
          ? { label: String(option), value: option }
          : option,
      )
```